src/NServiceBus.Persistence.AzureTable/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -181,23 +173,24 @@ public Task Complete(IContainSagaData sagaData, ISynchronizedStorageSession sess
         {
             var storageSession = (IWorkWithSharedTransactionalBatch)session;
             var meta = context.GetOrCreate<SagaInstanceMetadata>();
-            var sagaDataEntityToDelete = meta.Entities[sagaData.Id];
+            var sagaDataEntityToDeleteTuple = meta.Entities[sagaData.Id];
+            var sagaDataEntityToDelete = sagaDataEntityToDeleteTuple.Item2;
```

Comment:
It should be possible to deconstruct directly into the two values then you don't need to access the items